src/cr/cube/slices.py

```diff
@@ -0,0 +1,1140 @@
+# encoding: utf-8
+
+from __future__ import division
+
+import numpy as np
+from scipy.stats.contingency import expected_freq
+
+from cr.cube.util import lazyproperty
+from cr.cube.enum import DIMENSION_TYPE as DT
+
+
+class _CatXCatSlice(object):
+    def __init__(self, dimensions, counts, base_counts):
+        self._dimensions = dimensions
+        self._counts = counts
+        self._base_counts = base_counts
+
+    @lazyproperty
+    def _row_dimension(self):
+        return self._dimensions[0]
+
+    @lazyproperty
+    def _column_dimension(self):
+        return self._dimensions[1]
+
+    @lazyproperty
+    def _row_elements(self):
+        return self._row_dimension.valid_elements
+
+    @lazyproperty
+    def _column_elements(self):
+        return self._column_dimension.valid_elements
+
+    @lazyproperty
+    def _row_generator(self):
```

Comment:
You will probably find a more elegant way to do this, something with `iter` or yield. I didn't find it necessary for the POC work done here.